feat: add support for accessor arrays and refactor `stats/base/mediansorted`

vivekmaurya001 opened the pull request:
Resolves #{{TODO: add issue number}}.

This pull request:

-    add accessor arrays support and refactor `stats/base/mediansorted`

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   resolves #{{TODO: add issue number}} 

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

 - [x] Read, understood, and followed the [contributing guidelines](https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md)


* * *

@stdlib-js/reviewers 

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

vivekmaurya001 committed:
feat: add support for accessor arrays and refactor stats/base/mediansorted

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: passed
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: passed
  - task: lint_javascript_tests
    status: passed
  - task: lint_javascript_benchmarks
    status: passed
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: passed
  - task: lint_typescript_tests
    status: passed
  - task: lint_license_headers
    status: passed
---

kgryte commented on lib/node_modules/@stdlib/stats/base/mediansorted/benchmark/benchmark.js:
```diff
@@ -37,13 +45,7 @@ var mediansorted = require( './../lib/mediansorted.js' );
-	for ( i = 0; i < len; i++ ) {
-		x.push( i );
-	}
+	var x = uniform( len, 0, len, options );
```
@vivekmaurya001 This is no longer sorted data. To match the previous logic, use something like `array/base/linspace` and create an array of linearly spaced numbers.

vivekmaurya001 replied:
yess , i noticed it now it doesnt gives sorted , i Will correct it soon

vivekmaurya001 replied:
done

kgryte commented on lib/node_modules/@stdlib/stats/base/mediansorted/benchmark/benchmark.ndarray.js:
```diff
@@ -37,13 +45,7 @@ var mediansorted = require( './../lib/ndarray.js' );
-	for ( i = 0; i < len; i++ ) {
-		x.push( i );
-	}
+	var x = uniform( len, 0, len, options );
```
Same comment.

vivekmaurya001 replied:
ook

vivekmaurya001 replied:
done

kgryte commented on lib/node_modules/@stdlib/stats/base/mediansorted/README.md:
```diff
@@ -134,16 +126,12 @@ var v = mediansorted.ndarray( N, x, 2, 1 );
-for ( i = 0; i < x.length; i++ ) {
-    x[ i ] = i - 5.0;
-}
+var x = uniform( 10, -5, 5, {
```
This is no longer sorted. Use array/linspace instead.

vivekmaurya001 replied:
done

kgryte commented on lib/node_modules/@stdlib/stats/base/mediansorted/examples/index.js:
```diff
@@ -18,16 +18,12 @@
-for ( i = 0; i < x.length; i++ ) {
-	x[ i ] = i - 5.0;
-}
+var x = uniform( 10, -5, 5, {
```
Same comment as README

vivekmaurya001 replied:
done

kgryte reviewed (changes requested):
Other than some nits regarding generated data, this is looking good.

vivekmaurya001 committed:
feat: add support for accessor arrays and refactor stats/base/mediansorted

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: passed
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: passed
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: passed
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: na
---

vivekmaurya001 committed:
Update accessors.js

done some comment changes

Signed-off-by: Vivek maurya <155618190+vivekmaurya001@users.noreply.github.com>

kgryte commented on lib/node_modules/@stdlib/stats/base/mediansorted/README.md:
```diff
@@ -134,16 +126,10 @@ var v = mediansorted.ndarray( N, x, 2, 1 );
-for ( i = 0; i < x.length; i++ ) {
-    x[ i ] = i - 5.0;
-}
+var x = linspace( -5, 5, 10 );
```
```suggestion
var x = linspace( -5.0, 5.0, 10 );
```

kgryte commented on lib/node_modules/@stdlib/stats/base/mediansorted/benchmark/benchmark.js:
```diff
@@ -37,13 +38,7 @@ var mediansorted = require( './../lib/mediansorted.js' );
-	for ( i = 0; i < len; i++ ) {
-		x.push( i );
-	}
+	var x = linspace( 0, len, len );
```
```suggestion
	var x = linspace( 0.0, len, len );
```

kgryte commented on lib/node_modules/@stdlib/stats/base/mediansorted/benchmark/benchmark.ndarray.js:
```diff
@@ -37,13 +38,7 @@ var mediansorted = require( './../lib/ndarray.js' );
-	for ( i = 0; i < len; i++ ) {
-		x.push( i );
-	}
+	var x = linspace( 0, len, len );
```
```suggestion
	var x = linspace( 0.0, len, len );
```

kgryte commented on lib/node_modules/@stdlib/stats/base/mediansorted/examples/index.js:
```diff
@@ -18,16 +18,10 @@
-for ( i = 0; i < x.length; i++ ) {
-	x[ i ] = i - 5.0;
-}
+var x = linspace( -5, 5, 10 );
```
```suggestion
var x = linspace( -5.0, 5.0, 10 );
```

kgryte committed:
Apply suggestions from code review

Signed-off-by: Athan <kgryte@gmail.com>

kgryte reviewed (approved):
LGTM
